server-master/src/main/scala/com/aliyun/emr/rss/service/deploy/master/Master.scala

```diff
@@ -421,32 +421,34 @@ private[deploy] class Master(
     context.reply(OneWayMessageResponse)
   }
 
-  private def handleGetClusterLoadStatus(context: RpcCallContext): Unit = {
-    val (_, _, _, result) = getClusterLoad
+  private def handleGetClusterLoadStatus(context: RpcCallContext, numPartitions: Int): Unit = {
+    val clusterSlotsUsageLimit: Double = RssConf.clusterSlotsUsageLimitPercent(conf)
+    val (totalSlots, usedSlots, _) = getClusterLoad(numPartitions)
+
+    val totalUsedRatio: Double = (usedSlots + numPartitions) / totalSlots.toDouble
+    val result = totalUsedRatio >= clusterSlotsUsageLimit
+    logInfo(s"Current cluster slots usage:$totalUsedRatio, conf:$clusterSlotsUsageLimit, " +
+        s"overload:$result")
     context.reply(GetClusterLoadStatusResponse(result))
   }
 
-  private def getClusterLoad: (Int, Int, Int, Boolean) = {
+  private def getClusterLoad(numPartitions: Int = 0): (Int, Int, Int) = {
     if (workersSnapShot.isEmpty) {
```

Comment:
workersSnapShot is a method and will return a new object each invocation. I suggest to assign to a variable here.

Reply:
Thanks, I have retrieved a variable [workers] keeping [workersSnapShot] result